terrain_navigation_ros/src/terrain_planner.cpp

```diff
@@ -760,12 +760,15 @@ void TerrainPlanner::publishPositionHistory(rclcpp::Publisher<nav_msgs::msg::Pat
 void TerrainPlanner::publishGlobalPositionSetpoints(
     rclcpp::Publisher<mavros_msgs::msg::GlobalPositionTarget>::SharedPtr pub, const double latitude,
     const double longitude, const double altitude, const Eigen::Vector3d &velocity, const double curvature) {
+  // bit mask for path following: ignore force_set, yaw, yaw_rate
+  const uint16_t path_mask = 0b1111111000000000;
+
   using namespace mavros_msgs;
   // Publishes position setpoints sequentially as trajectory setpoints
   mavros_msgs::msg::GlobalPositionTarget msg;
   msg.header.stamp = this->get_clock()->now();
   msg.coordinate_frame = mavros_msgs::msg::GlobalPositionTarget::FRAME_GLOBAL_REL_ALT;
-  msg.type_mask = 0.0;
+  msg.type_mask = path_mask;
```

Comment:
```suggestion
  msg.type_mask = mavros_msgs::msg::GlobalPositionTarget::FORCE | mavros_msgs::msg::GlobalPositionTarget::IGNORE_YAW | IGNORE_YAW_RATE;
```

Reply:
Good suggestion @Jaeyoung-Lim - I need to check what happens to the highest bits in the mask using this approach - may need to update AP's handling as well to ensure a match.

Reply:
Interesting, this is part of the mavlink standard no? Or is it something special to the ardupilot dialect?

Reply:
AP does not use the POSITION_TARGET_TYPEMASK flags to construct the checks, instead uses a hardcoded `unint16_t` bit pattern (as was done here). Just need to check the AND logic used in the case evaluation there works as expected.